Sources/NIO/ChannelPipeline.swift

```diff
@@ -173,6 +173,23 @@ public final class ChannelPipeline: ChannelInvoker {
         return promise.futureResult
     }
 
+    /// Adds the provided channel handlers to the pipeline in the order given, taking account
+    /// of the behaviour of `ChannelHandler.add(first:)`.
+    public func addHandlers(_ handlers: [ChannelHandler], first: Bool) -> EventLoopFuture<Void> {
+        var handlers = handlers
+        if first {
+            handlers = handlers.reversed()
+        }
+
+        return EventLoopFuture<Void>.andAll(handlers.map { add(handler: $0) }, eventLoop: eventLoop)
+    }
+
+    /// Adds the provided channel handlers to the pipeline in the order given, taking account
+    /// of the behaviour of `ChannelHandler.add(first:)`.
+    public func addHandlers(_ handlers: ChannelHandler..., first: Bool) -> EventLoopFuture<Void> {
+        return addHandlers(handlers, first: first)
+    }
+
```

Comment:
Cool, happy with the placement. Two notes: can we add the API docs for `-parameters` and `-returns`, and can we put this in an extension in this file rather than in the main declaration? It just helps keep these visually distinct in the codebase. 👍

Reply:
done